### PR TITLE
Fix the --include-related option for 10.9

### PR DESCRIPTION
**Pull request by danieroux**
CXXFLAGS='-std=c++11' need to be set for xapian to be happy, it seems.

Fix from: https://github.com/djcb/mu/issues/380

### CODE REVIEW

File: Library/Formula/mu.rb [hunk 1]
```diff
@@ -43,6 +43,10 @@ def install
     # shipped by default with Mac OS X is too old.
     ENV["EMACS"] = "no" if build.without? "emacs"
 
+    # Make --related work on 10.9
+    # https://github.com/djcb/mu/issues/380
+    ENV.append 'CXXFLAGS', '-std=c++11'
```
Comment (mikemcquaid) [inline Library/Formula/mu.rb]: What about 10.8?

Comment (danieroux) [inline Library/Formula/mu.rb]: It works on 10.8, and I believe will continue working.

The line can be wrapped in `if version >= 10.9` - I'm not finding what that line will look like in brew.

Comment (mikemcquaid) [inline Library/Formula/mu.rb]: 10.8 doesn't ship with a libc++. If this needs C++11 instead use `ENV.cxx11` and/or `needs :cxx11`. Thanks!

**Comment by danieroux** [comment]
Updated pull request.

Thanks @mikemcquaid for the hint, I found the guard I wanted. This will now continue to work for 10.8. And finally >= 10.9 will actually work.

**Comment by tdsmith** [comment]
Does using a different optimization level work, per https://github.com/djcb/mu/issues/380#issuecomment-42462635? That sounds like a more parsimonious solution. You can add `ENV.O2` to the top of `install` to check.

**Comment by danieroux** [comment]
@tdsmith ENV.O0 does the trick. This exercise also gave me a good excuse to add a decent brew test in there.

File: Library/Formula/mu.rb [hunk 2]
```diff
@@ -63,4 +66,33 @@ def caveats; <<-EOS.undent
...
+      From: "Wile E. Coyote" <wile@example.com>
+      To: "Road Runner" <fasterthanyou@example.com>
+      Date: Mon, 4 Aug 2008 12:40:49 +0200
+      Message-id: <2222222222@example.com>
+      References: <1111111111@example.com>
+
+      This used to happen outdoors. It was more fun then.
+    EOS
+
+    system "#{bin}/mu index --muhome #{testpath} --maildir=#{testpath}"
```
Comment (mikemcquaid) [inline Library/Formula/mu.rb]: Split the `system` arguments `"like", "this"`

File: Library/Formula/mu.rb [hunk 3]
```diff
@@ -63,4 +66,33 @@ def caveats; <<-EOS.undent
...
+
+    assert_equal "1", `#{bin}/mu find --muhome #{testpath} msgid:2222222222@example.com | wc -l`.strip
```
Comment (mikemcquaid) [inline Library/Formula/mu.rb]: Something like `assert_equal 1, shell_output("#{bin}/mu ...").lines.size` would be preferable, thanks!

**Comment by danieroux** [comment]
@mikemcquaid thank you for the style check :-)

File: Library/Formula/mu.rb [hunk 4]
```diff
@@ -63,4 +66,40 @@ def caveats; <<-EOS.undent
...
+                        "--muhome",
+                        testpath,
+                        "--maildir=#{testpath}"
+
+    mu_find = "#{bin}/mu find --muhome #{testpath} "
+    find_message = "#{mu_find} msgid:2222222222@example.com"
+    find_message_and_related = "#{mu_find} --include-related msgid:2222222222@example.com"
+
+    assert_equal 1, shell_output(find_message).lines.size
+    assert_equal 2, shell_output(find_message_and_related).lines.size,
```
Comment (mikemcquaid) [inline Library/Formula/mu.rb]: Sorry, my bad, `.size` should be `.count` for Ruby 1.8 on OS X 10.8.

**Comment by danieroux** [comment]
@mikemcquaid this PR is done done now.

**Comment by mikemcquaid** [comment]
@danieroux Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format is `mu: fix --include-related option for 10.9`.

**Pull request closed without merging**